### ISSUE TITLE
Disable running of "git tag -l" during build.

### ISSUE THREAD

**Issue by jameycribbs** [palantir/godel#355]
I'm trying to use godel in my gitlab-ci.yml file, but during the run of godelw build, it errors out when trying to run "git tag -l", with:

Error: command [git tag -l] failed with output fatal: Not a git repository (or any of the parent directories): .git

Is there a way to tell godlew build to NOT run git tag?  I looked through all the docs but did not see that option for the dist.yml file.

**Comment by nmiyake** [comment]
Thanks for the feedback, this is very helpful.

Right now there is not a way to do this -- having a notion of a `version` is baked into a lot of parts of this, and right now that makes a `git` call.

An earlier release candidate made the logic for determining a version configurable as a script -- re-introducing that feature should solve this, as you could specify a script that returns an empty string/hard-coded version and that would bypass all of the `git` calls.

I pulled that feature because I didn't know if it would be used and also thought that, if we were going to make that pluggable, it might be better to provide that as an asset rather than as just a script.

Given that you ran into this, I think this is worth fixing. I'll update this ticket when I've made the fix with instructions on how to configure a project to incorporate the fix.

**Comment by nmiyake** [comment]
@jamesneale just pushed a fix for this that's available in 2.0.0-rc10 (2.0.0-rc9 can also make use of the fix by manually taking the disgo 1.0.0-rc13 release).

Add the following configuration to the `godel/config/dist-plugin.yml` file:

```yaml
project-versioner:
  type: script
  config:
    script: |
            #!/usr/bin/env bash
            echo "my-version"
```

The content of the `script` key will be run as a script, and the output that it produces (that is, prints to STDOUT) will be used as the version for the project. In this example, it hard-codes "my-version". However, if the version is available as an environment variable (or you have some other way to compute it), you can add that logic as needed.

The script inherits the environment variables of the process, but also gets an environment variable named "PROJECT_DIR" whose value is the absolute path to the project directory.

**Comment by jameycribbs** [comment]
Thank you very much!
